hash/hash.go

```diff
@@ -19,6 +19,13 @@ var DefaultHash = sha256.New
 // size is a default size for hashing algorithm.
 var size = DefaultHash().Size()
 
+func checkLen(data []byte) error {
+	if len(data) != size {
```

Comment:
if this is hash.Hahs length it should be in hash package , what do you think?

Reply:
This function is used with different parameters: 2 with directly `Hash`, 2 with `[]byte`.
So it doesn't make sense to put it as a struct function (`func (h *Hash) checkLen(data []byte) error {`).
> it should be in hash package

it's already in hash package.